llbutil/secrets.go

```diff
@@ -0,0 +1,89 @@
+package llbutil
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/earthly/earthly/secretsclient"
+
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/secrets"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// ErrNoSecretsClient occurs when the secrets client is referenced but was never provided
+var ErrNoSecretsClient = fmt.Errorf("no secrets client provided")
+
+type secretProvider struct {
+	store  secrets.SecretStore
+	client secretsclient.Client
+}
+
+// Register registers the secret provider
+func (sp *secretProvider) Register(server *grpc.Server) {
+	secrets.RegisterSecretsServer(server, sp)
+}
+
+func (sp *secretProvider) getSecretFromServer(path string) ([]byte, error) {
+	if sp.client == nil {
+		return nil, ErrNoSecretsClient
+	}
+	data, err := sp.client.Get(path)
```

Comment:
We should cache this locally (possibly in an in-memory daemon that keeps running?). This is probably a future PR though.

Reply:
that's a good plan, I opened https://github.com/earthly/earthly/issues/397 to track it.